kclvm/error/src/lib.rs

```diff
@@ -379,24 +404,15 @@ impl ParseError {
         let span = match self {
             ParseError::UnexpectedToken { span, .. } => span,
             ParseError::Message { span, .. } => span,
+            ParseError::String { span, .. } => span,
         };
         let loc = sess.sm.lookup_char_pos(span.lo());
         let pos: Position = loc.into();
-        let suggestions = match self {
-            ParseError::Message {
-                fix_info: Some(ref info),
-                ..
-            } => Some(vec![
-                info.suggestion
-                    .clone()
-                    .unwrap_or_else(|| "No suggestion available".to_string()),
-                info.replacement.clone().unwrap_or_else(|| " ".to_string()),
-            ]),
+        let (start_pos, end_pos) = self.generate_modified_range(&pos);
```

Comment:
Why do we need to calculate modified_range separately? Is this different from the range by span in line 404?